sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_container_registry_client.py

```diff
@@ -743,6 +750,59 @@ async def update_tag_properties(
             repository=repository
         )
 
+    @distributed_trace_async
+    async def upload_blob(self, repository: str, data: IO, **kwargs) -> Tuple[str, int]:
+        """Upload an artifact blob.
+
+        :param str repository: Name of the repository.
+        :param data: The blob to upload. Note: This must be a seekable stream.
+        :type data: IO
+        :returns: The digest and size of the uploaded blob.
+        :rtype: Tuple[str, int]
+        :raises ValueError: If the parameter repository or data is None.
+        """
+        try:
+            start_upload_response_headers = await cast(
+                Dict[str, str],
+                self._client.container_registry_blob.start_upload(
+                    repository, cls=_return_response_headers, **kwargs
+                )
+            )
+            digest, location, blob_size = await self._upload_blob_chunk(
+                start_upload_response_headers['Location'], data, **kwargs
+            )
+            complete_upload_response_headers = await cast(
+                Dict[str, str],
```

Comment:
You can't await a `Dict`, so you would either need to move the `await` statement into the case statement, or cast the result as `Awaitable[Dict[str, str]]`